lib/read_only/dl_esm_inf/read_only.f90

```diff
@@ -0,0 +1,416 @@
+                print *,"PSYDATA: checked variable ", trim(name)
+            endif
+        else
+            this%checksums(this%next_var_index) = value
+        endif
+        this%next_var_index = this%next_var_index + 1
+    end subroutine ChecksumScalarInt
+
+    ! -------------------------------------------------------------------------
+    !> This subroutine declares a scalar single precision value.
+    !! @param[inout] this The instance of the read_only_verify_PSyDataType.
+    !! @param[in] name The name of the variable (string).
+    !! @param[in] value The value of the variable.
+    subroutine DeclareScalarReal(this, name, value)
+        implicit none
+        class(read_only_verify_PSyDataType), intent(inout), target :: this
+        character(*), intent(in) :: name
+        real, intent(in) :: value
+    end subroutine DeclareScalarReal
+
+    ! -------------------------------------------------------------------------
+    !> This subroutine either computes a checksum or compares a checksum
+    !! (depending on this%verify_checksums)
+    !! @param[inout] this The instance of the read_only_verify_PSyDataType.
+    !! @param[in] name The name of the variable (string).
+    !! @param[in] value The value of the variable.
+    subroutine ChecksumScalarReal(this, name, value)
+        implicit none
+        class(read_only_verify_PSyDataType), intent(inout), target :: this
+        character(*), intent(in) :: name
+        real, intent(in) :: value
+        real :: orig_value
+        integer(kind=int64) :: chksum64
+        integer(kind=int32) :: chksum32
+
+        ! Type-case from real to 32-bit int (same size, so
+        ! no undefined bits in result)
+        chksum32 = transfer(value, chksum32)
+        ! Now assign to 64 bit, so we have a 64 bit checksum
+        chksum64 = chksum32
+
+        if(this%verify_checksums) then
+            if(this%checksums(this%next_var_index) /= chksum64) then
+                ! Convert back to 32 bit, and then cast to real:
+                chksum32 = this%checksums(this%next_var_index)
+                orig_value = transfer(chksum32, orig_value)
+                print *,"--------------------------------------"
+                print *,"Real variable ", name, " has been modified in ", &
+                    this%module_name," : ", this%region_name
+                print *,"Original value: ", orig_value
+                print *,"New value:      ", value
+                print *,"--------------------------------------"
+            else if(this%verbosity>1) then
+                print *,"PSYDATA: checked variable ", trim(name)
+            endif
+        else
+            this%checksums(this%next_var_index) = chksum64
+        endif
+        this%next_var_index = this%next_var_index + 1
+    end subroutine ChecksumScalarReal
+
+    ! -------------------------------------------------------------------------
+    !> This subroutine declares a scalar single precision value.
```

Comment:
Done.

Reply:
(though I am wondering if we should add a special flag 'no declaration' to prevent the creation of the declaration calls. That would be similar to what happens implicitly already in profile - no declare or provide calls are done if no variable is printed). Well, enhancement for later.